### ISSUE TITLE
Can you add midi out feature?

### ISSUE THREAD

**Issue by galfaroth** [chaosprint/glicol#100]
I'd wish to connect the signal's input to my midi synthesizer.

**Comment by BenLeadbetter** [comment]
Would you consider going straight for the MIDI 2.0 protocol? The technology and adoption is still pretty young but might be worth considering for the extended feature set. Also the future proofing factor.

**Comment by chaosprint** [comment]
> Would you consider going straight for the MIDI 2.0 protocol? The technology and adoption is still pretty young but might be worth considering for the extended feature set. Also the future proofing factor.

of course. I just looked at a few crates, and they don't have good support yet, but they are all in wip status. Do you have any recommended crates?

**Comment by BenLeadbetter** [comment]
I must admit, I'm the lead maintainer of the [midi2](https://crates.io/crates/midi2) crate and I'm looking for some early adoption clients.. But yeah you're right, the rust MIDI 2.0 space (and the MIDI 2.0 space in general, tbh) is still very young.

**Comment by BenLeadbetter** [comment]
And for the transport layer, there's not yet a platform independent solution. [midir](https://crates.io/crates/midir) doesn't support MIDI 2.0, yet for instance. MacOS is covered for MIDI 2.0 transport support via the [coremidi](https://crates.io/crates/coremidi) crate - so perhaps it would be a macOS only feature?

**Comment by chaosprint** [comment]
> And for the transport layer, there's not yet a platform independent solution. [midir](https://crates.io/crates/midir) doesn't support MIDI 2.0, yet for instance. MacOS is covered for MIDI 2.0 transport support via the [coremidi](https://crates.io/crates/coremidi) crate - so perhaps it would be a macOS only feature?

I think the specific implementation of midi input and output is glicol-cli and web ide.

The language level only needs to be responsible for reading special information.

This repo is responsible for language and audio synthesis.

I just submitted an issue in glicol-cli repo. https://github.com/glicol/glicol-cli/issues/39
